src/app/ReadClient.h

```diff
@@ -351,11 +351,17 @@ class ReadClient : public Messaging::ExchangeDelegate
      *  Re-activate an inactive subscription.
      *
      *  When subscribing to LIT-ICD and liveness timeout reached and OnResubscriptionNeeded returns
-     * CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT, the read client will move to the InactiveICDSubscription state and
-     * resubscription can be triggered via OnActiveModeNotification().
+     *  CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT, the read client will move to the InactiveICDSubscription state and
+     *  resubscription can be triggered via OnActiveModeNotification().
      *
      *  If the subscription is not in the `InactiveICDSubscription` state, this function will do nothing. So it is always safe to
-     * call this function when a check-in message is received.
+     *  call this function when a check-in message is received.
+     *
+     *  If the server sends out check-in message, and there is a active tracked active subscription in client side at the same time,
+     * it means current client does not realize this tracked subscription has gone, and we should forcibly timeout current
+     * subscription, and schedule a new one.
```

Comment:
This comment does not match what the API actually does, really.

Reply:
rewrite the comments, thanks